File: mahfazti/src/main/java/com/ayman/mahfazti/report/dto/CategoryBreakdownItem.java
package com.ayman.mahfazti.report.dto;

import java.math.BigDecimal;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CategoryBreakdownItem {

    private Long categoryId;
    private String categoryName;
    private String categoryIcon;
    private BigDecimal totalAmount;
    private double percentage;
}

File: mahfazti_mobile/lib/domain/entities/category_breakdown_item.dart
class CategoryBreakdownItem {
  final int categoryId;
  final String categoryName;
  final String? categoryIcon;
  final double totalAmount;
  final double percentage;

  const CategoryBreakdownItem({
    required this.categoryId,
    required this.categoryName,
    this.categoryIcon,
    required this.totalAmount,
    required this.percentage,
  });
}